### PR TITLE
Feat/ajv dependency injection

### LINKED ISSUE #605: JSONSchema formats are not handled by AJV in serverless-contracts
### Describe the bug

When using json-schema contracts with the `format` syntax of json-schema, AJV data validation fails along with lambda execution. 😬

✅ I managed to fix it locally by applying Ajv recommendations:
https://ajv.js.org/guide/formats.html

```js
const Ajv = require("ajv")
const addFormats = require("ajv-formats")

const ajv = new Ajv()
addFormats(ajv)
```

I can make a PR implementing this fix 🚀

### Reproduction

To reproduce: define a json-schema based contract with a `format` attribute. Data validation will fail.
```ts
contractId: { type: 'string', format: 'uuid' }
```

### System Info

```shell
System:
    OS: macOS 13.1
    CPU: (8) x64 Apple M1
    Memory: 26.32 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 14.18.1 - ~/.nvm/versions/node/v14.18.1/bin/node
    Yarn: 3.2.0 - /opt/homebrew/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v14.18.1/bin/npm
  Browsers:
    Chrome: 111.0.5563.110
    Safari: 16.2
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/swarmion/swarmion/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://www.swarmion.dev/docs/).
- [X] Check that there isn't [already an issue](https://github.com/swarmion/swarmion/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### LINKED ISSUE #606: Improve the error message when the returned value does not satisfy the return type of the handler
### Clear and concise description of the problem

When using `getHandler` with a contract that contains an enum, if the returned value does not satisfy the return type, then the error message is vague.
image

### Suggested solution

Would appreciate a message like:
```
2023-03-29T13:19:06.131Z	c6896a68-ad8c-4d79-8e6d-2a25f0694360	ERROR	[
  {
    instancePath: '/flightStatuses/0/flightStatus',
    schemaPath: '#/properties/flightStatuses/items/properties/flightStatus/enum',
    keyword: 'enum',
    params: { allowedValues: ['OnTime', 'Delayed', 'Rerouted'] },
    message: '"Diverted" must be equal to one of the allowed values'
  }
]
```

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/swarmion/swarmion/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://www.swarmion.dev/docs/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by Sc0ra**
Fix #605 
Fix #606

### CODE REVIEW

File: packages/serverless-contracts/src/contracts/apiGateway/features/lambdaHandler.ts [hunk 1]
```diff
@@ -80,8 +81,6 @@ export const getApiGatewayHandler =
     const outputValidators: Record<string, ValidateFunction | undefined> = {};
 
     if (validateInput === true || validateOutput === true) {
-      const ajv = new Ajv({ keywords: ['faker'] });
```
Comment (GuillaumeLagrange) [inline packages/serverless-contracts/src/contracts/apiGateway/features/lambdaHandler.ts:83]: Peut être rajouter un mot dans la doc sur "comment tester avec faker" vu que le consommateur a la main sur Ajv now

File: user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md [hunk 2]
```diff
@@ -250,13 +254,18 @@ const handler = getHandler(myContract)(async event => {
 });
 ```
 
+:::info
+Regarding the `ajvInstance` argument, we advice you to use a singleton instance of ajv that you define in a separate file. This way, you can use the same instance for all your contracts and middlewares.
```
Comment (GuillaumeLagrange) [inline user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md:258]: advise

File: user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/2-eventbridge.md [hunk 3]
```diff
@@ -80,23 +80,32 @@ With EventBridge contracts, you can generate a natively typed lambda handler wit
...
+  ajvInstance,
+)(async event => {
   const { message, userId } = event.detail; // natively typed with the correct keys
 
   // write your business logic
 });
 ```
 
+:::info
+Regarding the `ajvInstance` argument, we advice you to use a singleton instance of ajv that you define in a separate file. This way, you can use the same instance for all your contracts and middlewares.
```
Comment (GuillaumeLagrange) [inline user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/2-eventbridge.md:96]: advise

File: packages/serverless-contracts/src/contracts/apiGateway/features/lambdaHandler.ts [hunk 4]
```diff
@@ -49,6 +49,7 @@ export const getApiGatewayHandler =
     } = OutputType<Contract>,
   >(
     contract: Contract,
+    ajv: Ajv,
```
Comment (CorentinDoue) [inline packages/serverless-contracts/src/contracts/apiGateway/features/lambdaHandler.ts:52]: don't you want to make it optional with a default value `new Ajv({ keywords: ['faker'] });`?

1. to minimize the breaking change
2. to simplify the usage if the default ajv configuration is suficient

**Review by GuillaumeLagrange — APPROVED** [review]

File: packages/serverless-contracts/src/contracts/apiGateway/features/lambdaHandler.ts (hunk 4 above)
Comment (GuillaumeLagrange) [reply packages/serverless-contracts/src/contracts/apiGateway/features/lambdaHandler.ts:52]: the tradeoff of this is that  Ajv would no longer be an optional peerDeps as planned but is still a dependency.

Comment (Sc0ra) [reply packages/serverless-contracts/src/contracts/apiGateway/features/lambdaHandler.ts:52]: To align with other features (getAxiosRequest, getFetchRequest), I would prefer to introduce a breaking change to align them on the same basis.
The breaking change documentation is quite straightforward, and I really think that it's a good practice for user to have the control on their ajv instance.

What do you think ?

File: user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md [hunk 5]
```diff
@@ -251,7 +252,12 @@ const handler = getHandler(myContract)(async event => {
 ```
 
 :::info
-By default, the `getHandler` feature will validate both the input and the output of your lambda. If you wish to disable one of those, you can use the optional second argument in the `getHandler` feature:
+Regarding the `ajv`option, we advise you to use a singleton instance of ajv that you define in a separate file. This way, you can use the same instance for all your contracts and middlewares.
```
Comment (adriencaccia) [inline user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md:255]:
```suggestion
Regarding the `ajv` option, we advise you to use a singleton instance of ajv that you define in a separate file. This way, you can use the same instance for all your contracts and middlewares.
```

File: user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md [hunk 6]
```diff
@@ -251,7 +252,12 @@ const handler = getHandler(myContract)(async event => {
...
+:::
```
Comment (adriencaccia) [inline user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md:256]: This line is useless

```suggestion
```

File: user-docs/documentation/docs/how-to-guides/4-migration-guides/5-ajv-dependency-injection.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
+---
+sidebar_position: 5
+---
+
+# `AJV` dependency injection in getHandler feature
```
Comment (adriencaccia) [inline user-docs/documentation/docs/how-to-guides/4-migration-guides/5-ajv-dependency-injection.md:5]:
```suggestion
# `AJV` dependency injection in `getHandler` feature
```

**Review by adriencaccia — CHANGES_REQUESTED** [review]

File: user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md (hunk 6 above)
Comment (Sc0ra) [reply user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md:256]: Je viens de tester, et ça casse tout si je l'enlève

**Review by adriencaccia — APPROVED** [review]

**Pull request merged**